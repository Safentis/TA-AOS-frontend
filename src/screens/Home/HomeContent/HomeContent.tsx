import React, { FC } from 'react';

import Character from '../../../components/Character/Character';

import './HomeContent.css';
import { Props } from './HomeContent.interface';
import { Entry } from '../../../App/App.interface';
import { useGlobalContext } from '../../../App/App.context';

// prettier-ignore
const HomeContent: FC<Props> = ({ className = '' }): React.ReactElement => {
  const { entries } = useGlobalContext();

  const ENTRIES = entries.map(
    ({ gender, status, image, name, species }: Entry, index: number) => (
      <Character
        className="content__character"
        gender={gender}
        species={species}
        status={status}
        image={image}
        name={name}
        key={index}
      />
    ),
  );

  return <div className={`content ${className}`}>{ENTRIES}</div>;
};

export default HomeContent;
