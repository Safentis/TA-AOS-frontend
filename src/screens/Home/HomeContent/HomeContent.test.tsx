import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import HomeContent from './HomeContent';

import { Props } from './HomeContent.interface';

const setUp = (props: Props) => {
  return shallow(<HomeContent {...props} />);
};

describe('<HomeContent />', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;
  let props: Props = {
    className: 'custom-style',
};

  beforeEach(() => {
    component = setUp(props);
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
