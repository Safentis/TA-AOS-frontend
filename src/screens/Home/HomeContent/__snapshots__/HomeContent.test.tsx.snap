// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeContent /> component does match to the snapshot 1`] = `
<div
  className="content custom-style"
/>
`;
